set-web/src/main/java/eu/europeana/set/web/service/impl/UserSetServiceImpl.java

```diff
@@ -66,6 +68,9 @@ public UserSet storeUserSet(UserSet newUserSet, Authentication authentication) t
 
 	// store in mongo database
 	updateTotal(newUserSet);
+	
+	checkDuplicateUserSets(newUserSet, false);
```

Comment:
this check applies only to entity best items sets and it is a validation. Therefore it should be called at the appropriate place within the BaseUserSetServiceImpl.validateEntityBestItemsSet()

Reply:
Improved.